WebAutopark.DAL/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Data;
+using Dapper;
+using Microsoft.Data.SqlClient;
+
+namespace WebAutopark.DAL.Repositories
+{
+    public abstract class BaseRepository<T>
+    {
+        string connectionString = null;
+        protected BaseRepository(string conn)
+        {
+            connectionString = conn;
+        }
+        protected void CreateBase(T instance, string sqlQueryString)
+        {
+            using (IDbConnection db = new SqlConnection(connectionString))
+            {
+                var sqlQuery = sqlQueryString;
+                db.Execute(sqlQuery, instance);
+            }
+        }
+
+        protected void DeleteBase(int id, string sqlQueryString)
+        {
+            using (IDbConnection db = new SqlConnection(connectionString))
+            {
+                var sqlQuery = sqlQueryString;
+                db.Execute(sqlQuery, new { id });
+            }
+        }
+
+        protected T GetByIdBase(int id, string sqlQueryString)
+        {
+            using (IDbConnection db = new SqlConnection(connectionString))
+            {
+                return db.Query<T>(sqlQueryString, new { id }).FirstOrDefault();
+            }
+        }
+
+        protected IEnumerable<T> GetAllBase(string sqlQueryString)
+        {
+            using (IDbConnection db = new SqlConnection(connectionString))
+            {
+                return db.Query<T>(sqlQueryString).ToList();
+            }
+        }
+
+        protected void UpdateBase(T instance, string sqlQueryString)
+        {
+            using (IDbConnection db = new SqlConnection(connectionString))
+            {
+                var sqlQuery = sqlQueryString;
+                db.Execute(sqlQuery, instance);
+            }
+        }
+    }
+}
```

Comment:
I think it strange variant of BaseRepository.
In BaseRepository u create connection and use it in derived repositories.
```suggestion
namespace WebAutopark.DAL.Repositories
{
    public abstract class BaseRepository : IDisposable, IAsyncDisposable
    {
        protected readonly DbConnection connection;
        protected BaseRepository(IConnectionProvider connectionProvider)
        {
            connection = connectionProvider.GetConnection();
        }
        
        public void Dispose()
        {
            connection?.Dispose();
        }

        public ValueTask DisposeAsync()
        {
            return connection.DisposeAsync();
        }
    }
}
```
In my suggestion class BaseRepository used to create connection only using IConnectionProvider.
IConnectionProvider is a interface with one method DbConnection GetConnection().
Implement class with this interface and add to DI.
Dont worry about Disposing it, DI do it bi yourself.

Reply:
I have fixed it in the way described below, so I have avoided further changes in Repositories. Looks like applicable. Please, let me know if I am to use better your code
```
public abstract class RepositoryBase : IDisposable, IAsyncDisposable
    {
        protected readonly DbConnection connection;
        protected RepositoryBase(string connectionString)
        {
            connection = new SqlConnection(connectionString);
        }
        
        public void Dispose()
        {
            connection?.Dispose();
        }

        public ValueTask DisposeAsync()
        {
            return connection.DisposeAsync();
        }
    }
```